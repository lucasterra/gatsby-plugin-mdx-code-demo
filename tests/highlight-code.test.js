const highlight = require('../highlight-code');
const demoCode = `
// prettier-ignore-start
import React from 'react'
// I wanna add some comments, just to make sure it will also escape them

const StaticAppBar = () => {
  return (
    <div>This is my appbar, it is so cool, right?</div>
  );
}

export default StaticAppBar
// prettier-ignore-end
`;

describe('highlight code', () => {
  it('should return a string formatted to jsx', () => {
    expect(highlight('jsx', demoCode)).toMatchSnapshot();
  });
});
