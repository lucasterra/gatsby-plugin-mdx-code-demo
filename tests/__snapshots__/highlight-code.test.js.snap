// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight code should return a string formatted to jsx 1`] = `
"<div className=\\"gatsby-highlight\\"><pre className=\\"language-jsx\\"><code>{\\"\\\\n\\"}// prettier-ignore-start{\\"\\\\n\\"}import React from 'react'{\\"\\\\n\\"}// I wanna add some comments, just to make sure it will also escape them{\\"\\\\n\\"}{\\"\\\\n\\"}const StaticAppBar = () =&gt; {\\"{\\"}{\\"\\\\n\\"}{\\"  \\"}return ({\\"\\\\n\\"}{\\"    \\"}<span className=\\"token tag\\"><span className=\\"token tag\\"><span className=\\"token punctuation\\">&lt;</span>div</span><span className=\\"token punctuation\\">&gt;</span></span><span className=\\"token plain-text\\">This is my appbar, it is so cool, right?</span><span className=\\"token tag\\"><span className=\\"token tag\\"><span className=\\"token punctuation\\">&lt;/</span>div</span><span className=\\"token punctuation\\">&gt;</span></span>{\\"\\\\n\\"}{\\"  \\"});{\\"\\\\n\\"}{\\"}\\"}{\\"\\\\n\\"}{\\"\\\\n\\"}export default StaticAppBar{\\"\\\\n\\"}// prettier-ignore-end{\\"\\\\n\\"}</code></pre></div>
"
`;
